### ISSUE TITLE
Soft incompatibility with "tectonic" mod

### ISSUE THREAD

**Issue by radekr5** [TerraformersMC/Terrestria#328]
Terrestria is interfering with creation of underground rivers from tectonic, making them look rough with strainght horisontal walls. ( Terrestria  mod version 6.0.13 , tectonic version 2.2, on fabric 1.20.1 , 0.15.9)

**Comment by gniftygnome** [comment]
I didn't see any of this on the 1.20.4 versions of the mods.  What makes you think Terrestria is causing it?  At a guess, perhaps because you are seeing a Terrestria biome in debug when it happens?  If so, what biome?

**Comment by gniftygnome** [comment]
I think I may have found an example.  I'm guessing this is happening where there are Caldera biomes, so you could disable those in Terrestria's config.  It could also happen with Canyon and Dunes.  Basically whatever Tectonic is doing to add the rivers probably happens before Minecraft's surface builders execute, and Terrestria uses surface builders to modify the surface height in some biomes.  There's not really anything Terrestria (actually, Biolith) can do differently here.  The noise surface has to already be generated before surface builders can run.

This isn't the only "soft" incompatibility between the two mods.  Tectonic modifies the depth of lukewarm oceans, which causes Terrestria's Volcanic Island biome to basically not generate (last I checked there isn't a good way to get the depth of ocean biomes from Minecraft).

**Comment by gniftygnome** [comment]
Yeah, Calderas generate pretty cursed in Tectonic worlds anyway.  You'll be better off just disabling them.

**Comment by Apollounknowndev** [comment]
> Basically whatever Tectonic is doing to add the rivers probably happens before Minecraft's surface builders execute

Yep, Tectonic handles underground rivers during terrain shaping which is before surface rules.

**Comment by gniftygnome** [comment]
I've thought about trying to move the surface builders earlier, because there are various downsides to using surface builders these days (particularly how Minecraft has already decided what Y level to place structures).  However, I suspect there would be some other downsides to doing it earlier and I haven't gotten around to investigating those yet.

**Comment by radekr5** [comment]
> I didn't see any of this on the 1.20.4 versions of the mods. What makes you think Terrestria is causing it? At a guess, perhaps because you are seeing a Terrestria biome in debug when it happens? If so, what biome?

aa, im using 1.20.1 version, maybe that . and as for what biome was on top af a river, it was a redwood forest ( with 4 thick and tall as F%^& trees).

So basicly, if i want to report this bug, i need to type to tectonic team about it ?

**Comment by radekr5** [comment]
> Yeah, Calderas generate pretty cursed in Tectonic worlds anyway. You'll be better off just disabling them.

i can do that via config file?( or mod menu mod) or need to dig deeper ?

**Comment by gniftygnome** [comment]
> So basicly, if i want to report this bug, i need to type to tectonic team about it ?

Apollo already commented here, and I doubt there's a way for Tectonic to make this work any easier than for me to.

> i can do that via config file?( or mod menu mod) or need to dig deeper ?

The biome config file file for Terrestria will be in the config dir of of the server (usually at the top level next to `mods` and `resourcepacks`, in `terrestria/biome.json`.  Just set `"caldera": true` to `"caldera": false`.

**Comment by radekr5** [comment]
> The biome config file file for Terrestria will be in the config dir of of the server (usually at the top level next to `mods` and 
>`resourcepacks`, in `terrestria/biome.json`. Just set `"caldera": true` to `"caldera": false`.

 okay, thanks for instructions ,  disabled caledras and volcanic island.

**Comment by gniftygnome** [comment]
BTW, Terrestria for Minecraft 1.21.2+ has a fix for the Volcanic Island issue when the ocean depth is modified.

The Calderas will still mess up underground rivers, however.
